src/bin/wasm-tools/component.rs

```diff
@@ -648,11 +648,20 @@ impl WitOpts {
                         dir.join("deps")
                     };
                     let packages_with_same_name = &names[&pkg.name.name];
+                    let packages_with_same_namespace = packages_with_same_name[&pkg.name.namespace];
                     let stem = if packages_with_same_name.len() == 1 {
-                        pkg.name.name.clone()
+                        if packages_with_same_namespace == 1 {
+                            pkg.name.name.clone()
+                        } else {
+                            let version_suffix = pkg
+                                .name
+                                .version
+                                .as_ref()
+                                .map(|ver| format!("@{}", ver))
```

Comment:
Nit: we're formatting here and below. If there's a way to avoid the extra format! call that would be nice.

Reply:
updated to format only if version exists.